src/codeu/chat/common/ConversationHeader.java

```diff
@@ -75,6 +77,8 @@ public ConversationHeader read(InputStream in) throws IOException {
 	// a hashmap with user id as key and their usertype as value
 	private HashMap<Uuid, UserType> userAccessRoles = new HashMap<Uuid, UserType>();
 
+	public LinkedList<String> bots = new LinkedList<String>();
```

Comment:
Basically, `LinkedList` is good if you expect to add or remove from the beginning of the list very often. In a `LinkedList` this is an `O(1)` operation, whereas in an `ArrayList` this is an `O(n)` operation (because all of the other elements have to shift down). On the flip side, `ArrayList` is good if you have to get elements from a specific index very often. In an `ArrayList` this is a `O(1)` operation, whereas in a LinkedList this is a `O(n)` operation (because you have to start from the front of the list and iterate over it until you get to the index).